src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -61,40 +62,34 @@ class PreviewIframe {
 
     private removeIframe() {
         const isCurrentIframePresent = document.body.contains(this.iframe)
-        if (!this.iframe || !isCurrentIframePresent) {
+        if (!isCurrentIframePresent) {
             return
         }
-        if (isCurrentIframePresent) {
-            document.body.removeChild(this.iframe)
-        }
-        if (this.iframe) {
-            this.iframe = null
-        }
-    }
 
-    private getIFrameByUrl(url: string): HTMLIFrameElement | null {
-        return document.querySelector(`iframe[src="${url}"]`)
+        document.body.removeChild(this.iframe)
     }
-    private getVisibleIframeByUrl(url: string): HTMLIFrameElement | null {
-        const iframe = this.getIFrameByUrl(url)
-        return iframe?.style.opacity === '1' ? iframe : null
+    /**
+     * HACK: needed because the instance is created thrice onload/toggle.
+     * Remove this check to see the issue
+     */
+    private getExisitingIframe(): HTMLIFrameElement | null {
+        return document.getElementById(this.iframeId) ? this.iframe : null
     }
 
     private initPreviewIframe() {
-        const url = Navigation.getPageUrl()
-        const existingIframe = this.getIFrameByUrl(url)
+        const url = Navigation.getDailyNotesUrl()
+        const existingIframe = this.getExisitingIframe()
         if (existingIframe) {
```

Comment:
I hope we'd be getting rid of this now. but if we were not  - I think the assignment would have been still required, to ensure that we operate on the appropriate underlying object

Reply:
this can be removed after #58 gets merged. I tested by merging with #57 already :)